src/components/PostDetails.tsx

```diff
@@ -1,105 +1,64 @@
+
+export const PostDetails: React.FC = () => {
+  const {
+    loaded,
+    hasError,
+    comments,
+  } = useAppSelector(state => state.comments);
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const dispatch = useAppDispatch();
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  useEffect(() => {
+    if (selectedPost?.id) {
+      dispatch(fetchComments(selectedPost?.id));
+    }
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
     setVisible(false);
-    setError(false);
+  }, [dispatch, selectedPost?.id]);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
+  const deleteComment = async (commentId: number) => {
+    dispatch(removeComment(commentId));
 
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
+    await commentsApi.deleteComment(commentId);
   };
 
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
-
-  const addComment = async ({ name, email, body }: CommentData) => {
+  const createComment = async ({ name, email, body }: CommentData) => {
     try {
       const newComment = await commentsApi.createComment({
         name,
         email,
         body,
-        postId: post.id,
+        postId: selectedPost?.id,
       });
 
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(addComment(newComment));
     } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(setError(true));
     }
   };
 
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
-
-    await commentsApi.deleteComment(commentId);
-  };
-
   return (
     <div className="content" data-cy="PostDetails">
       <div className="block">
         <h2 data-cy="PostTitle">
-          {`#${post.id}: ${post.title}`}
+          {`#${selectedPost?.id}: ${selectedPost?.title}`}
```

Comment:
Let's not render post details if there are no selected posts, either render it only if it's selected or show some kind of message that there are no details